app/src/main/java/Fragments/HomeFragment.java

```diff
@@ -22,11 +27,22 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public class HomeFragment extends Fragment {
+public class HomeFragment extends Fragment implements SearchView.OnQueryTextListener {
 	//Listing RecyclerView instances
+	//private View view;
+	//I chose to use an unfilteredlist to base filters off of. This way, the database is only called when something is changed in the database.
+	//Otherwise, every time the filter is changed, we would have to get the items from the database again.
+	private ArrayList<Listing> unfilteredList = new ArrayList<>();
 	private ArrayList<Listing> listingsList = new ArrayList<>();
 	private RecyclerView recyclerView;
 	private ListItemAdapter mAdapter;
+	private boolean isViewFiltered;
+	private String filterString;
+
+	// Fragment View
+	private Context context;
+
+	MenuItem fav;
```

Comment:
Is the variable fav for the favourite feature which will be implemented later?

Reply:
Unsure. @danielanthonyr ^?